bootloader/src/main.c

```diff
@@ -414,8 +417,19 @@ int main(void)
             // Initialize user application's Stack Pointer
             __set_MSP(*(__IO uint32_t*) ApplicationAddress);
 
-            // Set IWDG Timeout to 5 secs based on platform specific system flags
-            IWDG_Reset_Enable(5 * TIMING_IWDG_RELOAD);
+            uint8_t disable_iwdg = 0;
+#ifdef CHECK_FIRMWARE
+            // Pre-0.7.0 firmwares were expecting IWDG flag to be set in the DCT, now it's stored in
+            // the backup registers. As a workaround, we disable IWDG if an older firmware is detected
+            const uint16_t module_ver = get_main_module_version();
+            if (module_ver >= 0 && module_ver < SYSTEM_MODULE_VERSION_0_7_0_RC1) {
```

Comment:
@sergeuz Can we bound it also to the current SYSTEM_PART1_MODULE_VERSION, so if the header was corrupt and greater than SYSTEM_PART1_MODULE_VERSION (at the time bootloader was compiled), it would disable the IWDG as well.  That might mean updating to a new system firmware would disable the IWDG though... hmm, how about just giving it some kind of sane upper bounds for the time being like (SYSTEM_PART1_MODULE_VERSION + 500) to allow for some upgrade buffer but still protect against 99% of corrupt values `((65536-500)/65536)*100`.  If nothing else, we could protect against the erased state (0xFFFF) if that doesn't result in get_module_info() returning NULL but instead returning 65535 for `module_ver`.